src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,221 @@
+
+// Zero histogram bins
+void IntegerHistogram1D::zeroBins()
+{
+    bins_.clear();
+    nBinned_ = 0;
+    nMissed_ = 0;
+}
+
+// Return minimum value for data
+std::optional<int> IntegerHistogram1D::minimum() const { return minimum_; }
+
+// Return maximum value for data
+std::optional<int> IntegerHistogram1D::maximum() const { return maximum_; }
+
+// Bin specified value, returning success
+bool IntegerHistogram1D::bin(int x)
+{
+    if ((minimum_ && x < *minimum_) || (maximum_ && x > *maximum_))
+    {
+        nMissed_++;
+        return false;
+    }
+
+    bins_[x].first++;
+
+    return true;
+}
+
+// Return number of values binned over all bins
+long int IntegerHistogram1D::nBinned() const { return nBinned_; }
+
+// Accumulate current histogram bins into averages
+void IntegerHistogram1D::accumulate()
+{
+    for (auto &[key, value] : bins_)
+        value.second += (SampledDouble)value.first;
+
+    // Update accumulated data
+    updateAccumulatedData();
+}
+
+// Return histogram data
+std::map<int, std::pair<long int, SampledDouble>> &IntegerHistogram1D::bins() { return bins_; }
+const std::map<int, std::pair<long int, SampledDouble>> &IntegerHistogram1D::bins() const { return bins_; }
+
+// Add source histogram data into local array
+void IntegerHistogram1D::add(IntegerHistogram1D &other, int factor)
+{
+    if (bins_.size() != other.bins_.size())
+    {
+        Messenger::print("BAD_USAGE - Can't add IntegerHistogram1D data since arrays are not the same size ({} vs {}).\n",
+                         bins_.size(), other.bins_.size());
+        return;
+    }
+    for (auto &[key, value] : bins_)
+
+        bins_[value.first].first += other.bins_[value.first].first * factor;
+}
+
+// Return current data
+Data1D IntegerHistogram1D::data() const
+{
+    Data1D result = accumulatedData_;
+    for (auto &[key, value] : bins_)
+    {
+        result.values()[(key)] = value.first;
+    }
+    return result;
+}
+
+// Return accumulated (averaged) data
+const Data1D &IntegerHistogram1D::accumulatedData() const { return accumulatedData_; }
+
+/*
+ * Serialisation
+ */
+
+// Read data through specified LineParser
+bool IntegerHistogram1D::deserialise(LineParser &parser)
+{
+    clear();
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    std::optional<int> minLim = parser.argb(0) ? std::optional<int>(parser.argi(1)) : std::nullopt;
+    std::optional<int> maxLim = parser.argb(2) ? std::optional<int>(parser.argi(3)) : std::nullopt;
+    initialise(minLim, maxLim);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    nBinned_ = parser.argli(0);
+    nMissed_ = parser.argli(1);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+
+    auto nBins = parser.argli(0);
+
+    if (parser.getArgsDelim(LineParser::Defaults) != LineParser::Success)
+        return false;
+    for (int n; n <= nBins; n++)
```

Comment:
Your start value is undefined here which is a bit dangerous! Loop starts at zero so can use the condition `n < nBins`. Plus `auto` and the prefix `++`.
```suggestion
    for (auto n = 0; n < nBins; ++n)
```